modules/swiss/src/main/SwissApi.scala

```diff
@@ -435,14 +435,14 @@ final class SwissApi(
           .map(_.filter(_.user == userId))
           .flatMap { trophyList =>
             trophyList.headOption ?? { trophy =>
-              trophyApi.removeTrophiesByUrl(Swiss.swissUrl(swiss.id))
-              awardTrophies(swiss, trophy.date)
+              trophyApi.removeTrophiesByUrl(Swiss.swissUrl(swiss.id)) >>
```

Comment:
can you explain what are >> or >>- doing ?
I see no implementation of this operator in this particular file and removeTrophiesByUrl's return type is void so I feel puzzled
thank you

Reply:
They are defined here:
https://github.com/Mind-Sports-Games/lila/blob/44fb3db522b3c0b80d9001b8cf5bfe60b40365f7/modules/common/src/main/base/PimpedFutures.scala#L18
They are a little mind-bending and we had a conversation on discord recently about them which could be worth reading through - i'll have a dig for it